tests/test_constants_and_functions.cpp

```diff
@@ -148,4 +148,7 @@ TEST_SUBMODULE(constants_and_functions, m) {
             py::arg_v("y", 42, "<the answer>"),
             py::arg_v("z", default_value));
     });
+
+    // test noexcept(true) lambda
```

Comment:
If you think it's useful:

```
// test noexcept(true) lambda (#4565)
```

Very low effort here, much easier to make the connection later.